.github/workflows/check.yaml

```diff
@@ -140,3 +152,66 @@ jobs:
           export TEST_SNAP="$(pwd)/$(ls | grep '.*charmed-openstack-upgrader_.*\.snap$')"
           echo "$TEST_SNAP"
           tox -e func
+
+      - name: Upload Functional Test Coverage Artifact
+        uses: actions/upload-artifact@v4
+        with:
+          if-no-files-found: ignore
+          name: coverage-functional-${{ env.SYSTEM_ARCH }}
+          path: coverage-functional
+
+  tics-analysis:
+    runs-on: ubuntu-22.04
+    if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/main' }} || ${{ github.event_name == 'workflow_dispatch' }}
```

Comment:
For ` ${{ github.event_name == 'workflow_dispatch' }}`, it could be worth restricting to the main branch still? Otherwise this will trigger if someone manually runs the check workflow on a PR branch.